mlir/lib/Dialect/Linalg/Transforms/Hoisting.cpp

```diff
@@ -43,6 +43,121 @@ using llvm::dbgs;
 using namespace mlir;
 using namespace mlir::linalg;
 
+scf::ForOp replaceWithDifferentYield(RewriterBase &rewriter, scf::ForOp loop,
```

Comment:
If this function is a helper only used in this file, mark it as `static`.

In any case, please provide documentation.